backend/contract-service/src/main/java/org/egov/works/service/NotificationService.java

```diff
@@ -0,0 +1,336 @@
+        smsDetails.putAll(locationName);
+
+        return smsDetails;
+    }
+
+    private Map<String, List<String>> getOrgDetailsForCBOAdmin(ContractRequest request) {
+
+        RequestInfo requestInfo = request.getRequestInfo();
+        Contract contract = request.getContract();
+        String tenantId = contract.getTenantId();
+
+        // fetch project details - project name and location
+        List<LineItems> lineItems = request.getContract().getLineItems();
+        Map<String, List<LineItems>> lineItemsMap = lineItems.stream().collect(Collectors.groupingBy(LineItems::getEstimateId));
+        List<Estimate> estimates = estimateServiceUtil.fetchActiveEstimates(requestInfo, tenantId, lineItemsMap.keySet());
+        Map<String, String> projectDetails = projectServiceUtil.getProjectDetails(requestInfo, estimates.get(0));
+
+        Map<String,List<String>> orgDetails=organisationServiceUtil.getOrganisationInfo(request);
+        orgDetails.put("projectName", Collections.singletonList(projectDetails.get("projectName")));
+
+        return orgDetails;
+    }
+
+
+    private String getMessage(ContractRequest request, boolean isCBORole) {
+        //RETHINK-- NEED TO SEND TWO MESSAGES
+        Workflow workflow = request.getWorkflow();
+        String message = null;
+
+        if ("REJECT".equalsIgnoreCase(workflow.getAction()) && !isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_REJECT_LOCALIZATION_CODE);
+        } else if ("APPROVE".equalsIgnoreCase(workflow.getAction()) && !isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_APPROVE_CREATOR_LOCALIZATION_CODE);
+        } else if ("APPROVE".equalsIgnoreCase(workflow.getAction()) && isCBORole) {
+            message = getMessage(request, ContractServiceConstants.CONTRACTS_APPROVE_CBO_LOCALIZATION_CODE);
+
+        }
+
+        return message;
+    }
+
+    /**
+     * Gets the message from localization
+     *
+     * @param request
+     * @param msgCode
+     * @return
+     */
+    public String getMessage(ContractRequest request, String msgCode) {
+        String rootTenantId = request.getContract().getTenantId().split("\\.")[0];
+        Map<String, Map<String, String>> localizedMessageMap = getLocalisedMessages(request.getRequestInfo(), rootTenantId,
+                ContractServiceConstants.CONTRACTS_NOTIFICATION_ENG_LOCALE_CODE, ContractServiceConstants.CONTRACTS_MODULE_CODE);
+        return localizedMessageMap.get(ContractServiceConstants.CONTRACTS_NOTIFICATION_ENG_LOCALE_CODE + "|" + rootTenantId).get(msgCode);
+    }
+
+    /**
+     * Builds msg based on the format
+     *
+     * @param contract
+     * @param message
+     * @param userDetailsForSMS
+     * @return
+     */
+    public String buildMessageForRejectAction(Contract contract, Map<String, String> userDetailsForSMS, String message) {
+        message = message.replace("{CONTRACT_NUMBER}", contract.getContractNumber())
```

Comment:
Make below type of Strings as constants because It may cause memory issue and we are also using them multiple places
"projectName"

Reply:
these cannot be made constants for now. According to CK we are going to have customised message templates which will have projectName in different locations. So it has to be kept dynamic